### PR TITLE
localStorageStore now delete only items prefixed on version change

### PR DESCRIPTION
**Pull request by fzan**
fix issue on #8312

### CODE REVIEW

**Review by WiXSL — APPROVED** [review]

File: packages/ra-core/src/store/localStorageStore.ts [hunk 1]
```diff
@@ -83,7 +83,12 @@ export const localStorageStore = (
             if (localStorageAvailable) {
                 const storedVersion = getStorage().getItem(`${prefix}.version`);
                 if (storedVersion && storedVersion !== version) {
-                    getStorage().clear();
+                    const storage = getStorage();
+                    Object.keys(storage).forEach(key => {
+                        if (key.startsWith(`${prefix}`)) {
```
Comment (slax57) [inline packages/ra-core/src/store/localStorageStore.ts:88]: suggestion:

```suggestion
                        if (key.startsWith(prefix)) {
```

File: packages/ra-core/src/store/localStorageStore.spec.ts [hunk 2]
```diff
@@ -40,4 +40,14 @@ describe('localStorageStore', () => {
...
+    describe('reset preserving items outside Ra-Store', () => {
+        it('should reset the store', () => {
+            const store = localStorageStore();
+            store.setItem('foo', 'bar');
+            getStorage().setItem('baz', 'baz'); //set custom item in localstorage
+            store.reset();
+            expect(getStorage().getItem('baz')).toEqual('baz'); //expect not to be wiped on store reset
+            expect(store.getItem('foo')).toEqual(undefined);
+        });
+    });
```
Comment (slax57) [inline packages/ra-core/src/store/localStorageStore.spec.ts:52]: This test is good (and should be kept), but you are actually not testing the code you added. You should add a test about version change.

**Review by slax57 — CHANGES_REQUESTED** [review]

File: packages/ra-core/src/store/localStorageStore.ts (hunk 1 above)
Comment (fzan) [reply packages/ra-core/src/store/localStorageStore.ts:88]: oops, sorry my bad 😅

**Review by fzan — COMMENTED** [review]
More appropriate test case
Removed use of unneeded literal

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**